govcd/disk.go

```diff
@@ -0,0 +1,339 @@
+/*
+ * Copyright 2018 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"github.com/vmware/go-vcloud-director/types/v56"
+	"github.com/vmware/go-vcloud-director/util"
+	"net/http"
+	"net/url"
+)
+
+// Independent disk
+type Disk struct {
+	Disk   *types.Disk
+	client *Client
+}
+
+// Init independent disk struct
+func NewDisk(cli *Client) *Disk {
+	return &Disk{
+		Disk:   new(types.Disk),
+		client: cli,
+	}
+}
+
+// Create an independent disk in VDC
+// Reference: vCloud API Programming Guide for Service Providers vCloud API 30.0 PDF Page 102 - 103,
+// https://vdc-download.vmware.com/vmwb-repository/dcr-public/1b6cf07d-adb3-4dba-8c47-9c1c92b04857/
+// 241956dd-e128-4fcc-8131-bf66e1edd895/vcloud_sp_api_guide_30_0.pdf
+func (vdc *Vdc) CreateDisk(diskCreateParams *types.DiskCreateParams) ([]*types.Task, error) {
```

Comment:
The return type should be `Task` (as defined in `task.go`.
You can find a few dozen functions defined as such. For example, 
`func (vdc *Vdc) CreateOrgVDCNetwork(networkConfig *types.OrgVDCNetwork) (Task, error) `